### ISSUE TITLE
Lock Color

### ISSUE THREAD

**Issue by lucaorio** [lyft/coloralgorithm#9]
Is there a chance to see the _Lock Color_ feature part of Colorbox added here too, eventually? :)

**Comment by k-vyn** [comment]
Yep! Working on updating the algorithm and adding proper documentation.

**Comment by porada** [comment]
@k-vyn, is there any ETA on that?

**Comment by methyl** [comment]
@k-vyn I'd love to port https://www.colorbox.io/ directly as a Figma plugin and it will be great to have color locking possibility.

**Comment by macintoshhelper** [comment]
Hi, I've reverse engineered the color lock and minor steps feature from ColorBox, I just want to check that it's ok to publish it to my fork of this project? (licensed under the original license of-course) @k-vyn 

If so, let me know please, and I can add some docs, tests and examples, and open a PR to merge back to this repo 🙂.

**Comment by nahiyankhan** [comment]
@macintoshhelper Would love to check your additions! Do you have it public? 🙏

**Comment by macintoshhelper** [comment]
> @macintoshhelper Would love to check your additions! Do you have it public? 🙏

No, sorry, would need permission to use or publish it. I have an open PR with an ES6 refactor though: #13

**Comment by porada** [comment]
Since this repository is [open source](https://github.com/lyft/coloralgorithm/blob/3aae9b0f958db735b24b604ae3eb177859d4416f/package.json#L7), I don’t think you need explicit permission to publish a fork.

**Comment by nahiyankhan** [comment]
@macintoshhelper Understandable if thats the case! To @porada's point it seems like no permission is needed?

**Comment by macintoshhelper** [comment]
> Since this repository is [open source](https://github.com/lyft/coloralgorithm/blob/3aae9b0f958db735b24b604ae3eb177859d4416f/package.json#L7), I don’t think you need explicit permission to publish a fork.

Reverse engineered (decompiled code) does require explicit permission to use or publish.

I've already published a fork (ES6 + Rollup bundling): https://github.com/elemental-design/coloralgorithm/tree/feat/rollup-build – if there's interest, I can open it up for external contribution. I've been meaning to work on adding bezier curve interpolation (for multiple colour locks) for a while.

While it's not equivalent to color lock, I developed a library that automates hue ranges for you based on hex values (based on top of lyft/coloralgorithm): https://github.com/elemental-design/colors/blob/master/src/index.js .

**Comment by k-vyn** [comment]
@macintoshhelper @nahiyankhan 

Sorry for the long delay, I no longer work at Lyft, so I'm not sure who will take over the responsibility of this repo. This may become inactive. 

In any case, for context Lock hex feature was functionality that was built into ColorBox after many people requested it, and it was never re-added into the original Color Algorithm. 

I forked this repo, and I did a big revamp of the algorithm and provided a lot more functionality than the original one. Also, I hope it'll be a little easier to understand/develop with. 

Check it out here - https://github.com/k-vyn/coloralgorithm

**Comment by macintoshhelper** [comment]
> @macintoshhelper @nahiyankhan
> 
> Sorry for the long delay, I no longer work at Lyft, so I'm not sure who will take over the responsibility of this repo. This may become inactive.

No problem, thanks a lot for your reply and fork! 🙂 Would be great to have some place for community contribution to the project, I'd be happy to maintain a fork.

**Comment by nahiyankhan** [comment]
> @macintoshhelper @nahiyankhan
> 
> Sorry for the long delay, I no longer work at Lyft, so I'm not sure who will take over the responsibility of this repo. This may become inactive.
> 
> In any case, for context Lock hex feature was functionality that was built into ColorBox after many people requested it, and it was never re-added into the original Color Algorithm.
> 
> I forked this repo, and I did a big revamp of the algorithm and provided a lot more functionality than the original one. Also, I hope it'll be a little easier to understand/develop with.
> 
> Check it out here - https://github.com/k-vyn/coloralgorithm

No worries! You did an amazing work with this and it has helped me a lot! I was able to add in the lock functionality too actually. I agree with @macintoshhelper Would love to contribute. 

Good luck with your next venture!

**Comment by k-vyn** [comment]
Totally! I agree. I'm planning on maintaining that fork going forward, and I'd love contributions on that repo.
